api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto

```diff
@@ -183,4 +183,16 @@ message GrpcJsonTranscoder {
   // For a path with `/foo/first/bar/prefix/second/third/fourth`, `x=first`, `y=prefix/second`, `z=third/fourth`.
   // If this setting is not specified, the value defaults to :ref:`ALL_CHARACTERS_EXCEPT_RESERVED<envoy_api_enum_value_extensions.filters.http.grpc_json_transcoder.v3.GrpcJsonTranscoder.UrlUnescapeSpec.ALL_CHARACTERS_EXCEPT_RESERVED>`.
   UrlUnescapeSpec url_unescape_spec = 10 [(validate.rules).enum = {defined_only: true}];
+
+  // Whether to reject requests that cannot be transcoded.
+  //
+  // By default, the transcoder will silently pass through HTTP requests that all malformed.
+  // This includes requests with unknown query parameters, unregister paths, etc.
+  //
+  // Enable this flag to enable strict validation, resulting in the transcoder rejecting
+  // these requests with HTTP 400 Bad Request.
+  // gRPC requests will still silently pass through without transcoding.
+  //
+  // Defaults to false for backwards compatibility.
+  bool strict_http_request_validation = 11;
```

Comment:
Please add to the comment
// Please don't use this flag if upstream supports both HTTP and gRPC. 
// Only use this flag if upstream is gRPC and all its services have been specified in the descriptor file in the filter config.

Reply:
The benefit of this flag is:  proper error message,  without this flag,  it is connection reset error since grpc server could not handle http header.  It may cause retry if there is retry_on connection reset.

Reply:
Done.